### PR TITLE
[mypy] [9886] Reduce mypy errors in styles.py

### PR DESCRIPTION
**Pull request by rodrigc**
https://twistedmatrix.com/trac/ticket/9886

Eliminate mypy errors:

```
src/twisted/persisted/styles.py:87:49: error: Argument 3 to "pickle" has incompatible type "Callable[[Any, Any, Any], Any]"; expected
"Optional[Callable[[Union[Tuple[Callable[..., Type[MethodType]], Tuple[Any, ...]], Tuple[Callable[..., Type[MethodType]], Tuple[Any, ...], Optional[Any]]]], Type[MethodType]]]"  [arg-type]
    copy_reg.pickle(types.MethodType, pickleMethod, unpickleMethod)
                                                    ^
src/twisted/persisted/styles.py:231:48: error: Argument 3 to "pickle" has incompatible type "Callable[[Any, Any], Any]"; expected
"Optional[Callable[[Union[Tuple[Callable[..., Any], Tuple[Any, ...]], Tuple[Callable[..., Any], Tuple[Any, ...], Optional[Any]]]], Any]]"  [arg-type]
        copy_reg.pickle(OutputType, pickleStringO, unpickleStringO)
                                                   ^
src/twisted/persisted/styles.py:232:47: error: Argument 3 to "pickle" has incompatible type "Callable[[Any, Any], Any]"; expected
"Optional[Callable[[Union[Tuple[Callable[..., Any], Tuple[Any, ...]], Tuple[Callable[..., Any], Tuple[Any, ...], Optional[Any]]]], Any]]"  [arg-type]
        copy_reg.pickle(InputType, pickleStringI, unpickleStringI)
```

### CODE REVIEW

File: src/twisted/persisted/test/test_styles.py [hunk 1]
```diff
@@ -90,6 +102,16 @@ def test_instanceBuildingNameNotPresent(self):
...
 
 
+    def test_copyFunction(self):
+        """
+        Copying a function should exercise the pickle code
+        and preserve the function name
+        """
+        f = copy.copy(sampleFunction)
+        self.assertEqual(f, sampleFunction)
+        self.assertEqual(f.__name__, sampleFunction.__name__)
```
Comment (adiroiban) [inline src/twisted/persisted/test/test_styles.py]: This docstring is misleading.
From what I see in the implementation, it returns the exact same function... so is normal to have the same name.

**Review by adiroiban — CHANGES_REQUESTED** [review]
Thanks for the update.

I don't understand the new tests and docstring... it looks like there are there just to have a green test coverage

below is my suggestion

```diff
diff --git a/src/twisted/persisted/styles.py b/src/twisted/persisted/styles.py
index 5aaf4c13d..941e526ec 100644
--- a/src/twisted/persisted/styles.py
+++ b/src/twisted/persisted/styles.py
@@ -128,7 +128,7 @@ def _unpickleFunction(fullyQualifiedName):
 
 
 
-copy_reg.pickle(types.FunctionType, _pickleFunction, _unpickleFunction)
+copy_reg.pickle(types.FunctionType, _pickleFunction)
 
 
 
@@ -148,7 +148,7 @@ def unpickleModule(name):
 
 
 
-copy_reg.pickle(types.ModuleType, pickleModule, unpickleModule)
+copy_reg.pickle(types.ModuleType, pickleModule)
 
 
 
diff --git a/src/twisted/persisted/test/test_styles.py b/src/twisted/persisted/test/test_styles.py
index a3d3b83f4..45e7a905a 100644
--- a/src/twisted/persisted/test/test_styles.py
+++ b/src/twisted/persisted/test/test_styles.py
@@ -16,10 +16,14 @@ class Foo:
     """
     Helper class.
     """
+    def __init__(self):
+        self.instance_member = 'test-value'
+
     def method(self):
         """
         Helper method.
         """
+        return self.instance_member
 
 
 
@@ -81,13 +85,18 @@ class UnpickleMethodTests(unittest.TestCase):
 
     def test_instanceCopyMethod(self):
         """
-        Copying an instance method returns exercises the pickle code
-        and returns the instance method
+        Copying an instance method returns a new method with the same
+        behavior.
         """
         foo = Foo()
         m = copy.copy(foo.method)
+
         self.assertEqual(m, foo.method)
         self.assertIsNot(m, foo.method)
+        self.assertEqual('test-value', m())
+
+        foo.instance_member = 'new-value'
+        self.assertEqual('new-value', m())
 
 
     def test_instanceBuildingNameNotPresent(self):
@@ -104,12 +113,12 @@ class UnpickleMethodTests(unittest.TestCase):
 
     def test_copyFunction(self):
         """
-        Copying a function should exercise the pickle code
-        and preserve the function name
+        Copying a function returns the same reference, without creating
+        an actual copy.
         """
         f = copy.copy(sampleFunction)
-        self.assertEqual(f, sampleFunction)
-        self.assertEqual(f.__name__, sampleFunction.__name__)
+
+        self.assertIs(f, sampleFunction)
 
 
     def test_primeDirective(self):
```

File: src/twisted/persisted/test/test_styles.py (hunk 1 above)
Comment (rodrigc) [reply src/twisted/persisted/test/test_styles.py]: What's the suggestion here?

Comment (adiroiban) [reply src/twisted/persisted/test/test_styles.py]: `Copying a function returns the same reference, without creating an actual copy.` . is my suggestion from the diff

Comment (rodrigc) [reply src/twisted/persisted/test/test_styles.py]: OK, I incorporated your patch into this PR.

**Review by adiroiban — APPROVED** [review]
Looks good. thanks!

**Pull request merged**